### PR TITLE
Organize app examples

### LINKED ISSUE #16273: Organize the Examples Folder
### Outline & Motivation

Our examples folder is messy, it currently looks like this: https://github.com/Lightning-AI/lightning/tree/master/examples
It does not matter if you are an expert or a beginner, if you arrive here in this repository, the examples folder sticks out as an obvious place to start exploring. But it's a mess. Where to start? What is important? What is new, what is old? The Readme is outdated. 

### Pitch

Organize the top level folders into:

```
examples/
    app
    fabric
    pytorch
```

Organize the readme, either at the top level or in each subfolder, with sections for

Beginner
Intermediate
Expert

(or a similar structure organized by concepts). 

Delete examples that are too outdated or irrelevant. Make everything as concise as possible.

### Additional context

_No response_

cc @borda @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Moves the app examples under examples/app for easier navigation when exploring the examples folder.

Before:
```
examples/app_boring
examples/app_components
...
```

Now:

```
examples/app/boring
examples/app/components
...

```
Fixes #16273
Closes #17073

cc @borda @carmocca @akihironitta @justusschock @awaelchli

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -296,7 +296,7 @@ subprojects:
       - "src/lightning_app/*"
       - "tests/integrations_app/**"
       - "!tests/integrations_app/flagship/**"
-      - "examples/app_*/**"
+      - "examples/app/*/**"
```
Comment (carmocca) [inline .github/checkgroup.yml:299]:
```suggestion
      - "examples/app/**"
```

File: .github/checkgroup.yml [hunk 2]
```diff
@@ -296,7 +296,7 @@ subprojects:
...
       - "requirements/app/**"
       - "setup.py"
       - "!requirements/*/docs.txt"
```
Comment (carmocca) [inline .github/checkgroup.yml:302]: The app-examples workflow paths need to be updated too

Every change in the checkgroup paths should match a change in a workflows

**Review by carmocca — APPROVED** [review]

File: .github/checkgroup.yml (hunk 2 above)
Comment (awaelchli) [reply .github/checkgroup.yml:302]: Done

**Review by ethanwharris — APPROVED** [review]
LGTM 😃

**Review by Borda — APPROVED** [review]
btw, duplicated with #17073

**Pull request merged**